lib/src/rules/avoid_redundant_parameter_values.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2019, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Avoid redundant parameter values.';
+
+const _details = r'''Avoid redundant parameter values.
+
+**DON'T** declare parameters with values that match their defaults.
```

Comment:
Wouldn't be `arguments`.

Parameter: the thing the function AUTHOR defines.
Argument: the thing the function CALLER provides.

Right? @munificent who learned me this.

Reply:
Correct. The rule should probably be `avoid_redundant_argument_values`, and most uses of "parameter" in the comments should be "argument".

Reply:
Spot on.  Thanks for the catch!